generator/src/lib.rs

```diff
@@ -2144,6 +2198,23 @@ pub fn write_source_code(path: &Path) {
     let source_code = quote! {
         use std::fmt;
         use std::os::raw::*;
+        /// Iterates through the pointer chain. Includes the item that is passed into the function.
+        /// Stops at the last `BaseOutStructure` that has a null `p_next` field.
+        pub(crate) unsafe fn ptr_chain_iter<T>(
```

Comment:
Because it's static code, this should probably be defined in ash/lib.rs, not spliced into the generator output.